### PR TITLE
add level filter

### PR DESCRIPTION
**Pull request by anthonyseu**
screen shot 2017-03-10 at 12 26 29 am

### CODE REVIEW

File: components/TopicList.js [hunk 1]
```diff
@@ -1,36 +1,80 @@
...
             topics: props.topics
         };
+
+        this.onLevelFilterChange = this.onLevelFilterChange.bind(this);
     }
+
+    onLevelFilterChange(levelFilter) {
+        this.context.executeAction(TopicActions.loadTopics, {
+            params: {
+                searchBy: '',
```
Comment (anthonyseu) [inline components/TopicList.js]: @jacgrady1 how to combine the search bar and these filters

**Comment by anthonyseu** [comment]
@jacgrady1 @quiz1991

File: components/TopicList.js (hunk 1 above)
Comment (jacgrady1) [inline components/TopicList.js]: u mean query wise?

Comment (anthonyseu) [inline components/TopicList.js]: I mean how should I pass the search query to `TopicList` component, through props?

Comment (jacgrady1) [inline components/TopicList.js]: there are two ways to handle this . 
1. user input could trigger state. u could pass in state directly.
2. or you could write a store for this page. User input would trigger a flux flow. user input would eventually be in store.

File: components/TopicList.js [hunk 2]
```diff
@@ -1,36 +1,84 @@
...
+                levelFilter: levelFilter
+            }
+        });
+    }
+
     render() {
+        const levelFilters = [
+          {id: '0', label: 'Easy'},
+          {id: '1', label: 'Mid'},
+          {id: '2', label: 'Hard'}
```
Comment (jacgrady1) [inline components/TopicList.js]: would it be better to put them in a config file?

File: components/TopicList.js [hunk 3]
```diff
@@ -1,36 +1,84 @@
...
         this.state = {
-            topics: props.topics
+            topics: props.topics,
+            searchQuery: props.searchQuery
         };
+
+        this.onLevelFilterChange = this.onLevelFilterChange.bind(this);
     }
+
+    onLevelFilterChange(levelFilter) {
```
Comment (jacgrady1) [inline components/TopicList.js]: r u planning to combine all future filters into in function? if so, rename it to on Filters Change and pass in an object of filters?

File: components/TopicList.js (hunk 2 above)
Comment (anthonyseu) [inline components/TopicList.js]: We can put it on somewhere for constants.

File: components/TopicList.js (hunk 3 above)
Comment (anthonyseu) [inline components/TopicList.js]: This will be in my next PR to add category PR

Comment (jacgrady1) [inline components/TopicList.js]: okie

File: lib/levelFilter/constants.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default {
```
Comment (jacgrady1) [inline lib/levelFilter/constants.js]: any reasons this is not in config?

Comment (anthonyseu) [inline lib/levelFilter/constants.js]: Config is more fundamental, this is more on business logic side.

**Pull request merged**